Say/ViewController.swift

```diff
@@ -90,7 +93,29 @@ class ViewController: NSViewController {
             }
         }
     }
-
+    
+    @IBAction func selectText(_ sender: NSTextField){
+        let URLString = sender.stringValue
+        let url = URL(string: sender.stringValue)!
+        var data: NSData? = nil
+        data=NSData.init(contentsOf: url)//NSurlData
+        let dataString: String = String(data:data! as Data, encoding:String.Encoding.utf8)!
+        
+        let regex = try! NSRegularExpression(pattern: "<title>\\s*(.*)\\s*</title>", options: NSRegularExpression.Options())
+    
+        let result = regex.matches(in: dataString as String, options: NSRegularExpression.MatchingOptions(), range: NSRange(location: 0, length: dataString.lengthOfBytes(using: String.Encoding.utf8)))
+        
+        if result.count > 0 {
+            let range = result[0].rangeAt(1)
+            let text = (dataString as NSString).substring(with: range)
+            textView.string=text
+            print(text)
+            
+        }
+        
+    }
+    
```

Comment:
@Amanida's suggestion is great, but I will attach a little bit more code.
Even though the URL is correct, `NSData.init(contentsOf:)` still can fail.

```
    if let url = URL(string: sender.stringValue) {
        // if URL format is right
        if let data = NSData.init(contentsOf: url) {
            let dataString = String(data:data, encoding:String.Encoding.utf8)!
            if let result = findTitle(in: dataString) {
                textView.string = result
            } else {
                NSRunAlertPanel(@"URL fetching error", @"The page doesn't have a title.", @"OK", nil, nil);
            }
        } else {
            NSRunAlertPanel(@"URL fetching error", @"The URL is not accessible.", @"OK", nil, nil);
        }
    } else {
        NSRunAlertPanel(@"URL fetching error", @"The URL is malformed", @"OK", nil, nil);
    }
```